From 3b2839790ac0469de4190ff89966914af6b8bfae Mon Sep 17 00:00:00 2001
From: James Cowgill <james410@cowgill.org.uk>
Date: Tue, 12 May 2015 22:17:19 +0100
Subject: [PATCH 1/5] Remove direct dependencies to wxsqlite3 and sqlite3lib
 from all plugins

In CMake if you have a dependency chain which looks like:
  A -> B -> C
then CMake will automatically add the dependency A -> C for you.

This means that we can remove all the direct dependencies on wxsqlite3 and
sqlite3lib from all plugins if they're already linking against libcodelite.
---
 CMakePlugin/CMakeLists.txt                            | 2 --
 CallGraph/CMakeLists.txt                              | 2 +-
 CodeDesigner/CMakeLists.txt                           | 2 +-
 CodeFormatter/CMakeLists.txt                          | 2 --
 CodeLiteDiff/CMakeLists.txt                           | 2 --
 ContinuousBuild/CMakeLists.txt                        | 2 --
 Copyright/CMakeLists.txt                              | 2 --
 DatabaseExplorer/CMakeLists.txt                       | 2 --
 Debugger/CMakeLists.txt                               | 2 --
 ExternalTools/CMakeLists.txt                          | 2 --
 Gizmos/CMakeLists.txt                                 | 2 --
 HelpPlugin/CMakeLists.txt                             | 2 --
 LLDBDebugger/CMakeLists.txt                           | 2 --
 LLDBDebugger/codelite-lldb/CMakeLists.txt             | 2 --
 LiteEditor/CMakeLists.txt                             | 6 +++---
 MacBundler/CMakeLists.txt                             | 2 --
 MemCheck/CMakeLists.txt                               | 2 +-
 Outline/CMakeLists.txt                                | 2 --
 Plugin/CMakeLists.txt                                 | 6 +++---
 QmakePlugin/CMakeLists.txt                            | 2 --
 Runtime/templates/gizmos/CMakeLists.txt.plugin.wizard | 2 --
 SFTP/CMakeLists.txt                                   | 2 --
 SnipWiz/CMakeLists.txt                                | 2 --
 SpellChecker/CMakeLists.txt                           | 2 --
 Subversion2/CMakeLists.txt                            | 2 --
 Tweaks/CMakeLists.txt                                 | 2 --
 UnitTestCPP/CMakeLists.txt                            | 2 --
 WebTools/CMakeLists.txt                               | 2 --
 WordCompletion/CMakeLists.txt                         | 2 --
 ZoomNavigator/CMakeLists.txt                          | 2 --
 abbreviation/CMakeLists.txt                           | 2 --
 codelite_make/CMakeLists.txt                          | 2 +-
 codelitephp/CMakeLists.txt                            | 4 ----
 cppchecker/CMakeLists.txt                             | 2 --
 cscope/CMakeLists.txt                                 | 2 --
 git/CMakeLists.txt                                    | 2 --
 wxformbuilder/CMakeLists.txt                          | 2 --
 37 files changed, 10 insertions(+), 74 deletions(-)

diff --git a/CMakePlugin/CMakeLists.txt b/CMakePlugin/CMakeLists.txt
index bcba22e..eddb29d 100644
--- a/CMakePlugin/CMakeLists.txt
+++ b/CMakePlugin/CMakeLists.txt
@@ -65,8 +65,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/CallGraph/CMakeLists.txt b/CallGraph/CMakeLists.txt
index b058ff5..3dba7fd 100644
--- a/CallGraph/CMakeLists.txt
+++ b/CallGraph/CMakeLists.txt
@@ -48,7 +48,7 @@ add_library(${PLUGIN_NAME} SHARED ${SRCS})
 
 # Remove the "lib" prefix from the plugin name
 set_target_properties(${PLUGIN_NAME} PROPERTIES PREFIX "")
-target_link_libraries(${PLUGIN_NAME} ${LINKER_OPTIONS} ${wxWidgets_LIBRARIES} -L"${CL_LIBPATH}" -lwxsqlite3 -lsqlite3lib -llibcodelite -lplugin)
+target_link_libraries(${PLUGIN_NAME} ${LINKER_OPTIONS} ${wxWidgets_LIBRARIES} -L"${CL_LIBPATH}" libcodelite plugin)
 # Make sure that the plugin will not start build before 'plugin.so' is ready
 add_dependencies(${PLUGIN_NAME} plugin)
 
diff --git a/CodeDesigner/CMakeLists.txt b/CodeDesigner/CMakeLists.txt
index 26377b0..40679c7 100644
--- a/CodeDesigner/CMakeLists.txt
+++ b/CodeDesigner/CMakeLists.txt
@@ -33,4 +33,4 @@ add_library(${PLUGIN_NAME} SHARED ${SRCS})
 
 # Remove the "lib" prefix from the plugin name
 set_target_properties(${PLUGIN_NAME} PROPERTIES PREFIX "")
-target_link_libraries(${PLUGIN_NAME} ${LINKER_OPTIONS} ${wxWidgets_LIBRARIES} -L"${CL_LIBPATH}" -lwxsqlite3 -lsqlite3lib -llibcodelite -lplugin)
+target_link_libraries(${PLUGIN_NAME} ${LINKER_OPTIONS} ${wxWidgets_LIBRARIES} -L"${CL_LIBPATH}" libcodelite plugin)
diff --git a/CodeFormatter/CMakeLists.txt b/CodeFormatter/CMakeLists.txt
index 4aa6ad0..e447950 100644
--- a/CodeFormatter/CMakeLists.txt
+++ b/CodeFormatter/CMakeLists.txt
@@ -53,8 +53,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 
 
diff --git a/CodeLiteDiff/CMakeLists.txt b/CodeLiteDiff/CMakeLists.txt
index bdc1104..121d2a1 100644
--- a/CodeLiteDiff/CMakeLists.txt
+++ b/CodeLiteDiff/CMakeLists.txt
@@ -61,8 +61,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 # Installation destination
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/ContinuousBuild/CMakeLists.txt b/ContinuousBuild/CMakeLists.txt
index 3af6e63..d37c37a 100644
--- a/ContinuousBuild/CMakeLists.txt
+++ b/ContinuousBuild/CMakeLists.txt
@@ -53,7 +53,5 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/Copyright/CMakeLists.txt b/Copyright/CMakeLists.txt
index 72ca60d..09e1722 100644
--- a/Copyright/CMakeLists.txt
+++ b/Copyright/CMakeLists.txt
@@ -53,7 +53,5 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/DatabaseExplorer/CMakeLists.txt b/DatabaseExplorer/CMakeLists.txt
index 8500b67..b6c8e16 100644
--- a/DatabaseExplorer/CMakeLists.txt
+++ b/DatabaseExplorer/CMakeLists.txt
@@ -100,8 +100,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${GTK_LIBS}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       wxshapeframework
                       databaselayersqlite
                       )
diff --git a/Debugger/CMakeLists.txt b/Debugger/CMakeLists.txt
index 0a6a15a..413ed2e 100644
--- a/Debugger/CMakeLists.txt
+++ b/Debugger/CMakeLists.txt
@@ -53,8 +53,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 
 CL_INSTALL_DEBUGGER(${PLUGIN_NAME})
diff --git a/ExternalTools/CMakeLists.txt b/ExternalTools/CMakeLists.txt
index 74375ee..1071308 100644
--- a/ExternalTools/CMakeLists.txt
+++ b/ExternalTools/CMakeLists.txt
@@ -54,8 +54,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 # Installation destination
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
\ No newline at end of file
diff --git a/Gizmos/CMakeLists.txt b/Gizmos/CMakeLists.txt
index 113b80d..603054e 100644
--- a/Gizmos/CMakeLists.txt
+++ b/Gizmos/CMakeLists.txt
@@ -52,8 +52,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 # Installation destination
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
\ No newline at end of file
diff --git a/HelpPlugin/CMakeLists.txt b/HelpPlugin/CMakeLists.txt
index 5e2b577..4788085 100644
--- a/HelpPlugin/CMakeLists.txt
+++ b/HelpPlugin/CMakeLists.txt
@@ -57,8 +57,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 
 # Use CodeLite's macro: CL_INSTALL_PLUGIN which handles both OSX and Linux installation
diff --git a/LLDBDebugger/CMakeLists.txt b/LLDBDebugger/CMakeLists.txt
index 935874b..25cfb47 100644
--- a/LLDBDebugger/CMakeLists.txt
+++ b/LLDBDebugger/CMakeLists.txt
@@ -94,8 +94,6 @@ endif()
                               ${wxWidgets_LIBRARIES}
                               libcodelite
                               plugin
-                              wxsqlite3 
-                              sqlite3lib
                               )
         # Installation destination
         CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/LLDBDebugger/codelite-lldb/CMakeLists.txt b/LLDBDebugger/codelite-lldb/CMakeLists.txt
index 163af51..5c5c906 100644
--- a/LLDBDebugger/codelite-lldb/CMakeLists.txt
+++ b/LLDBDebugger/codelite-lldb/CMakeLists.txt
@@ -26,8 +26,6 @@ target_link_libraries(  codelite-lldb
                         LLDBProtocol
                         libcodelite
                         plugin
-                        wxsqlite3 
-                        sqlite3lib
                     )
 
 CL_INSTALL_EXECUTABLE(codelite-lldb)
diff --git a/LiteEditor/CMakeLists.txt b/LiteEditor/CMakeLists.txt
index fd23f83..92f7f57 100644
--- a/LiteEditor/CMakeLists.txt
+++ b/LiteEditor/CMakeLists.txt
@@ -91,11 +91,11 @@ else (UNIX)
 endif (UNIX)
 
 if(GTK2_FOUND)
-    target_link_libraries(codelite ${LINKER_OPTIONS} ${GTK2_LIBRARIES} ${wxWidgets_LIBRARIES} ${CLANG_LIBRARY} -L"${CL_LIBPATH}" -lwxsqlite3 -lsqlite3lib -llibcodelite -lplugin ${ADDITIONAL_LIBRARIES})
+    target_link_libraries(codelite ${LINKER_OPTIONS} ${GTK2_LIBRARIES} ${wxWidgets_LIBRARIES} ${CLANG_LIBRARY} -L"${CL_LIBPATH}" libcodelite plugin ${ADDITIONAL_LIBRARIES})
 elseif (GTK3_FOUND)
-    target_link_libraries(codelite ${LINKER_OPTIONS} ${GTK3_LIBRARIES} ${wxWidgets_LIBRARIES} ${CLANG_LIBRARY} -L"${CL_LIBPATH}" -lwxsqlite3 -lsqlite3lib -llibcodelite -lplugin ${ADDITIONAL_LIBRARIES})
+    target_link_libraries(codelite ${LINKER_OPTIONS} ${GTK3_LIBRARIES} ${wxWidgets_LIBRARIES} ${CLANG_LIBRARY} -L"${CL_LIBPATH}" libcodelite plugin ${ADDITIONAL_LIBRARIES})
 else()
-    target_link_libraries(codelite ${LINKER_OPTIONS} ${wxWidgets_LIBRARIES} ${CLANG_LIBRARY} -L"${CL_LIBPATH}" -lwxsqlite3 -lsqlite3lib -llibcodelite -lplugin ${ADDITIONAL_LIBRARIES})
+    target_link_libraries(codelite ${LINKER_OPTIONS} ${wxWidgets_LIBRARIES} ${CLANG_LIBRARY} -L"${CL_LIBPATH}" libcodelite plugin ${ADDITIONAL_LIBRARIES})
 endif()
 
 ##################################################################
diff --git a/MacBundler/CMakeLists.txt b/MacBundler/CMakeLists.txt
index 98cd991..17fcd41 100644
--- a/MacBundler/CMakeLists.txt
+++ b/MacBundler/CMakeLists.txt
@@ -45,8 +45,6 @@ set_target_properties(${PLUGIN_NAME} PROPERTIES PREFIX "")
 target_link_libraries(${PLUGIN_NAME} 
                       ${LINKER_OPTIONS} 
                       ${wxWidgets_LIBRARIES} 
-                      wxsqlite3 
-                      sqlite3lib 
                       libcodelite 
                       plugin)
 
diff --git a/MemCheck/CMakeLists.txt b/MemCheck/CMakeLists.txt
index 6c89bcd..e5723ee 100644
--- a/MemCheck/CMakeLists.txt
+++ b/MemCheck/CMakeLists.txt
@@ -55,7 +55,7 @@ if(UNIX AND NOT APPLE)
 
     # Remove the "lib" prefix from the plugin name
     set_target_properties(${PLUGIN_NAME} PROPERTIES PREFIX "")
-    target_link_libraries(${PLUGIN_NAME} ${LINKER_OPTIONS} ${wxWidgets_LIBRARIES} -L"${CL_LIBPATH}" -lwxsqlite3 -lsqlite3lib -llibcodelite -lplugin)
+    target_link_libraries(${PLUGIN_NAME} ${LINKER_OPTIONS} ${wxWidgets_LIBRARIES} -L"${CL_LIBPATH}" libcodelite plugin)
 
     # Make sure that the plugin will not start build before 'plugin.so' is ready
     add_dependencies(${PLUGIN_NAME} plugin)
diff --git a/Outline/CMakeLists.txt b/Outline/CMakeLists.txt
index 2c5f1e4..2483bb5 100644
--- a/Outline/CMakeLists.txt
+++ b/Outline/CMakeLists.txt
@@ -55,8 +55,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 # Installation destination
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
\ No newline at end of file
diff --git a/Plugin/CMakeLists.txt b/Plugin/CMakeLists.txt
index cf8caaf..ade4992 100644
--- a/Plugin/CMakeLists.txt
+++ b/Plugin/CMakeLists.txt
@@ -93,11 +93,11 @@ FILE(GLOB SRCS "*.cpp")
 # Define the output
 add_library(plugin SHARED ${SRCS})
 if(GTK2_FOUND)
-    target_link_libraries(plugin ${LINKER_OPTIONS} ${ADDITIONAL_LIBRARIES} ${GTK2_LIBRARIES} ${wxWidgets_LIBRARIES} -L"${CL_LIBPATH}" -lwxsqlite3 -lsqlite3lib -llibcodelite)
+    target_link_libraries(plugin ${LINKER_OPTIONS} ${ADDITIONAL_LIBRARIES} ${GTK2_LIBRARIES} ${wxWidgets_LIBRARIES} -L"${CL_LIBPATH}" libcodelite)
 elseif (GTK3_FOUND)
-    target_link_libraries(plugin ${LINKER_OPTIONS} ${ADDITIONAL_LIBRARIES} ${GTK3_LIBRARIES} ${wxWidgets_LIBRARIES} -L"${CL_LIBPATH}" -lwxsqlite3 -lsqlite3lib -llibcodelite)
+    target_link_libraries(plugin ${LINKER_OPTIONS} ${ADDITIONAL_LIBRARIES} ${GTK3_LIBRARIES} ${wxWidgets_LIBRARIES} -L"${CL_LIBPATH}" libcodelite)
 else()
-    target_link_libraries(plugin ${LINKER_OPTIONS} ${ADDITIONAL_LIBRARIES} ${wxWidgets_LIBRARIES} -L"${CL_LIBPATH}" -lwxsqlite3 -lsqlite3lib -llibcodelite)
+    target_link_libraries(plugin ${LINKER_OPTIONS} ${ADDITIONAL_LIBRARIES} ${wxWidgets_LIBRARIES} -L"${CL_LIBPATH}" libcodelite)
 endif()
 
 if (NOT MINGW)
diff --git a/QmakePlugin/CMakeLists.txt b/QmakePlugin/CMakeLists.txt
index ee49d39..2187bfc 100644
--- a/QmakePlugin/CMakeLists.txt
+++ b/QmakePlugin/CMakeLists.txt
@@ -53,8 +53,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 # Installation destination
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/Runtime/templates/gizmos/CMakeLists.txt.plugin.wizard b/Runtime/templates/gizmos/CMakeLists.txt.plugin.wizard
index a11d69b..15b7447 100644
--- a/Runtime/templates/gizmos/CMakeLists.txt.plugin.wizard
+++ b/Runtime/templates/gizmos/CMakeLists.txt.plugin.wizard
@@ -57,8 +57,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 
 # Use CodeLite's macro: CL_INSTALL_PLUGIN which handles both OSX and Linux installation
diff --git a/SFTP/CMakeLists.txt b/SFTP/CMakeLists.txt
index 55e3e9e..79d5f9e 100644
--- a/SFTP/CMakeLists.txt
+++ b/SFTP/CMakeLists.txt
@@ -54,8 +54,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 # Installation destination
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/SnipWiz/CMakeLists.txt b/SnipWiz/CMakeLists.txt
index b808dce..2d82f0c 100644
--- a/SnipWiz/CMakeLists.txt
+++ b/SnipWiz/CMakeLists.txt
@@ -54,7 +54,5 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/SpellChecker/CMakeLists.txt b/SpellChecker/CMakeLists.txt
index 26a71dd..abc9dbf 100644
--- a/SpellChecker/CMakeLists.txt
+++ b/SpellChecker/CMakeLists.txt
@@ -83,8 +83,6 @@ else ()
                           ${LIBHUNSPELL}
                           libcodelite
                           plugin
-                          wxsqlite3 
-                          sqlite3lib
                           )
 
     # Installation destination
diff --git a/Subversion2/CMakeLists.txt b/Subversion2/CMakeLists.txt
index b7db282..b8c461f 100644
--- a/Subversion2/CMakeLists.txt
+++ b/Subversion2/CMakeLists.txt
@@ -54,8 +54,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 
 # Installation destination
diff --git a/Tweaks/CMakeLists.txt b/Tweaks/CMakeLists.txt
index b0ead78..8ceb94c 100644
--- a/Tweaks/CMakeLists.txt
+++ b/Tweaks/CMakeLists.txt
@@ -55,7 +55,5 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/UnitTestCPP/CMakeLists.txt b/UnitTestCPP/CMakeLists.txt
index 0f90690..cc004d7 100644
--- a/UnitTestCPP/CMakeLists.txt
+++ b/UnitTestCPP/CMakeLists.txt
@@ -55,8 +55,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 
 
diff --git a/WebTools/CMakeLists.txt b/WebTools/CMakeLists.txt
index 4b4ed3e..6510750 100644
--- a/WebTools/CMakeLists.txt
+++ b/WebTools/CMakeLists.txt
@@ -57,8 +57,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/WordCompletion/CMakeLists.txt b/WordCompletion/CMakeLists.txt
index 1f90843..a397c81 100644
--- a/WordCompletion/CMakeLists.txt
+++ b/WordCompletion/CMakeLists.txt
@@ -53,7 +53,5 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
\ No newline at end of file
diff --git a/ZoomNavigator/CMakeLists.txt b/ZoomNavigator/CMakeLists.txt
index f39f153..5779819 100644
--- a/ZoomNavigator/CMakeLists.txt
+++ b/ZoomNavigator/CMakeLists.txt
@@ -54,7 +54,5 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
\ No newline at end of file
diff --git a/abbreviation/CMakeLists.txt b/abbreviation/CMakeLists.txt
index 37645a8..d472cf0 100644
--- a/abbreviation/CMakeLists.txt
+++ b/abbreviation/CMakeLists.txt
@@ -52,8 +52,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 # Installation destination
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/codelite_make/CMakeLists.txt b/codelite_make/CMakeLists.txt
index ff96fa1..f75a6e1 100644
--- a/codelite_make/CMakeLists.txt
+++ b/codelite_make/CMakeLists.txt
@@ -42,6 +42,6 @@ if ( UNIX )
 endif ( UNIX )
 
 target_link_libraries(codelite-make ${LINKER_OPTIONS} ${wxWidgets_LIBRARIES} -L"${CL_LIBPATH}" 
-                        wxsqlite3 sqlite3lib libcodelite plugin ${UTIL_LIB} )
+                        libcodelite plugin ${UTIL_LIB} )
 
 CL_INSTALL_EXECUTABLE(codelite-make)
diff --git a/codelitephp/CMakeLists.txt b/codelitephp/CMakeLists.txt
index b1e94df..ed4e416 100644
--- a/codelitephp/CMakeLists.txt
+++ b/codelitephp/CMakeLists.txt
@@ -94,8 +94,6 @@ target_link_libraries(${PLUGIN_NAME}
                       PHPParser
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 # Installation destination
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
@@ -106,8 +104,6 @@ target_link_libraries(PHPUnitTests
                      -L"${CMAKE_LIBRARY_OUTPUT_DIRECTORY}" 
                      -L"${CL_LIBPATH}" 
                      PHPParser 
-                     wxsqlite3 
-                     sqlite3lib 
                      libcodelite 
                      plugin 
                      ${ADDITIONAL_LIBRARIES} 
diff --git a/cppchecker/CMakeLists.txt b/cppchecker/CMakeLists.txt
index 97bb57f..2d6e265 100644
--- a/cppchecker/CMakeLists.txt
+++ b/cppchecker/CMakeLists.txt
@@ -48,8 +48,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
 
diff --git a/cscope/CMakeLists.txt b/cscope/CMakeLists.txt
index d1ba3b9..847b0c1 100644
--- a/cscope/CMakeLists.txt
+++ b/cscope/CMakeLists.txt
@@ -53,7 +53,5 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/git/CMakeLists.txt b/git/CMakeLists.txt
index 55222b4..660f7b6 100644
--- a/git/CMakeLists.txt
+++ b/git/CMakeLists.txt
@@ -48,8 +48,6 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
diff --git a/wxformbuilder/CMakeLists.txt b/wxformbuilder/CMakeLists.txt
index c1c4f76..4036482 100644
--- a/wxformbuilder/CMakeLists.txt
+++ b/wxformbuilder/CMakeLists.txt
@@ -53,7 +53,5 @@ target_link_libraries(${PLUGIN_NAME}
                       ${wxWidgets_LIBRARIES}
                       libcodelite
                       plugin
-                      wxsqlite3 
-                      sqlite3lib
                       )
 CL_INSTALL_PLUGIN(${PLUGIN_NAME})
-- 
2.1.4

